Scripts/aws_stack_provision.ps1

```diff
@@ -53,76 +106,61 @@ if ($EXISTING_INSTANCE_COUNT -eq 1 )
       Write-Host "VM is in stopped state"
       Write-Host "Starting the VM"
       $NEW_STATUS = (((Start-EC2Instance -InstanceID $INSTANCE_ID).CurrentState).Name).Value
-      (Get-SECSecretValue -SecretId privatekey/AzureDevOps).SecretString > $env:tmp\key.pem
-      $RetryCount = 5
-      while ( ((Get-EC2PasswordData -InstanceId $INSTANCE_ID -PemFile $env:tmp\key.pem) -eq $null ) -and ($RetryCount -gt 0) )
-      {
-         Start-Sleep -Seconds 120
-         $RetryCount -= 1
-         Write-Host "Waiting to fetch VM password"
-      }
-
-      if ( $RetryCount -le 0)
+      $RETRY_COUNT = fetch_vm_password $INSTANCE_ID
+      if ( $RETRY_COUNT -le 0)
       {
-         throw "Timeout: 10 minutes expired waiting to fetch VM password"
+         throw "Timeout: 20 minutes expired in waiting to fetch the VM password"
       }
 
       Write-Host "Able to fetch the VM password"
-      Write-Host "Current status of VM is $NEW_STATUS"
+      Write-Host "Current state of VM is $NEW_STATUS"
    }
 
    else
    {
-      Write-Host "VM is neither Running nor stopped"
-      throw "VM status is $INSTANCE_STATE"
+      Write-Host "VM is neither in Running State nor in Stopped State"
+      throw "VM state is $INSTANCE_STATE"
    }
 }
 
 elseif ($EXISTING_INSTANCE_COUNT -eq 0){
-   Write-Host "No Exisiting VM with Lansa Version tag = $lansa_version found"
+   Write-Host "No Exisiting VM found with Lansa Version tag = $lansa_version"
    $NO_OF_AMIS = (Get-EC2Image -Filter @{ Name="tag:LansaVersion"; Values=$clone_lansa_version } | Select-Object ImageId | Measure-Object | Select-Object Count).count
    if ($NO_OF_AMIS -eq 1)
    {
-      Write-Host "Found 1 AMI with Lansa vesion tag = $lansa_version"
+      Write-Host "Found 1 AMI with Lansa vesion tag = $clone_lansa_version"
       $AMI_ID = (Get-EC2Image -Filter @{ Name="tag:LansaVersion"; Values=$clone_lansa_version }).ImageId
       $STACK_NAME = "DB-Regression-VM-" + $lansa_version
       try
       {
          $EXISTING_CFN_STACK_STATUS = ((Get-CFNStack -StackName $STACK_NAME).StackStatus).Value
-         throw "CFN Stack with stack name = $STACK_NAME exist and is in $EXISTING_CFN_STACK_STATUS state"
+         Write-Host "CFN Stack with stack name = $STACK_NAME exist and is in $EXISTING_CFN_STACK_STATUS state"
+         $RETRY_COUNT = remove_cfn_stack $STACK_NAME
+         if ( $RETRY_COUNT -le 0 )
+         {
+            throw "Timeout: 30 minutes expired waiting to Delete CFN Stack $STACK_NAME"
+         }
+         Write-Host "CFN Stack $STACK_NAME is deleted"
+         Get-CFNStack -StackName $STACK_NAME
       }
       catch [System.InvalidOperationException]
```

Comment:
Why are you using a `catch block` to implement non-error processing functionality? It looks like this code should be in a `finally block`. And maybe this `catch block` should do nothing. And maybe there needs to be a generic catch block added to handle errors - `throw`?